doc/notebooks/grass_jupyter.ipynb

```diff
@@ -30,46 +26,45 @@
    "metadata": {},
    "outputs": [],
    "source": [
+    "import os\n",
+    "import subprocess\n",
+    "import sys\n",
+    "\n",
+    "# Add GRASS to path\n",
     "gisbase = subprocess.check_output([\"grass\", \"--config\", \"path\"], text=True).strip()\n",
     "os.environ[\"GISBASE\"] = gisbase\n",
-    "sys.path.append(os.path.join(gisbase, \"etc\", \"python\"))"
-   ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": null,
-   "metadata": {},
-   "outputs": [],
-   "source": [
+    "sys.path.append(os.path.join(gisbase, \"etc\", \"python\"))\n",
+    "\n",
+    "# Import GRASS packages\n",
     "import grass.script as gs\n",
-    "import grass.jupyter as gj"
-   ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": null,
-   "metadata": {},
-   "outputs": [],
-   "source": [
+    "import grass.jupyter as gj\n",
+    "\n",
     "# Start GRASS Session\n",
-    "gj.init(\"../../data/grassdata\", \"nc_basic_spm_grass7\", \"user1\")"
-   ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": null,
-   "metadata": {},
-   "outputs": [],
-   "source": [
-    "# Set computational region to the study area.\n",
+    "gj.init(\"../../data/grassdata\", \"nc_basic_spm_grass7\", \"user1\")\n",
+    "\n",
+    "# Set computational region to the elevation raster.\n",
     "gs.run_command(\"g.region\", raster=\"elevation\")"
    ]
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "## Non-interactive Map Display"
+    "## GRASS Renderer\n",
+    "\n",
+    "The `GrassRenderer` class creates and displays GRASS maps as PNG images. There are two ways to add elements to the display. First, the name of the *GRASS* display module can be called as an attribute by replacing the \".\" with \"\\_\" in the module name. For example:\n",
+    "````\n",
+    "m = GrassRenderer()\n",
+    "m.d_rast(map=\"elevation\")\n",
+    "````\n",
+    "\n",
+    "Alternatively, *GRASS* display modules can be called with the `run()` method:\n",
+    "````\n",
+    "m = GrassRenderer()`\n",
```

Comment:
Extra backtick at the end of the line.

Reply:
I separated computation and visualization and switch to d_shade - that is a neat module!